src/map/packet_system.cpp

```diff
@@ -2378,29 +2379,133 @@ void SmallPacket0x042(map_session_data_t* const PSession, CCharEntity* const PCh
+        std::string login_message = luautils::GetServerMessage(msg_language);
+
+        PChar->pushPacket(new CServerMessagePacket(login_message, msg_language, msg_timestamp, msg_offset));
+        PChar->pushPacket(new CCharSyncPacket(PChar));
+
+        // TODO: kill player til theyre dead and bsod
+        const char* fmtQuery = "SELECT version_mismatch FROM accounts_sessions WHERE charid = %u";
+        int32       ret      = _sql->Query(fmtQuery, PChar->id);
+        if (ret != SQL_ERROR && _sql->NextRow() == SQL_SUCCESS)
         {
-            PChar->pushPacket(new CChatMessagePacket(PChar, CHAT_MESSAGE_TYPE::MESSAGE_SYSTEM_1, "Server does not support this client version."));
+            if ((bool)_sql->GetUIntData(0))
+            {
+                PChar->pushPacket(new CChatMessagePacket(PChar, CHAT_MESSAGE_TYPE::MESSAGE_SYSTEM_1, "Server does not support this client version."));
+            }
+        }
+    }
+    else if (msg_type == 2) // Fish Ranking Packet
+    {
+        // The Message Chunk acts as a "sub-type" for the request
+        // 1 = First packet of ranking table
+        // 2 = Subsequent packet of ranking table
+        // 10 = ???
+        // 11 = ??? Prepare to withdraw?
+        // 12 = Response to a fish submission (No ranking or score - both 0) - Before ranking
+        // 13 = Fish Rank Self, including the score and rank (???) following fish submission (How is it ranked??)
+        std::vector<FishingContestEntry*> entries;
+
+        uint8                maxFakes    = 15;
+        uint8                maxEntries  = 100;
+        uint8                realEntries = fishingcontest::FishingRankEntryCount();
+        uint8                fakeEntries = realEntries >= maxFakes ? 0 : (uint8)std::min((int)maxFakes, maxFakes - realEntries); // Max 15 Fakes
+        uint8                msg_entries = std::min(maxEntries, (uint8)(realEntries + fakeEntries));                             // Max 100 entries
+        FishingContestEntry* entry       = nullptr;
+        uint8                entryVal    = 0;
+        uint8                blockSize   = sizeof(FishingContestEntry); // Should be 36
+
+        // Add the "Self" block for 0x1C - Either player data, or empty, depending on the chunk
+        if (msg_chunk != 2)
+        {
+            // Client requesting the fish ranking menu header - All empty timestamps
+            // In either case, we need the "Fish Rank Self" block
+            entry = fishingcontest::GetPlayerEntry(PChar);
+            if (entry)
+            {
+                // For type 1, include the player's Fish Rank Self packet from the contest.  Otherwise, just use some current info.
+                if (msg_chunk == 1)
+                {
+                    entry->resultCount = msg_entries;
+                }
+                else
+                {
+                    FishingContestEntry* original = entry;
+                    entry                         = new FishingContestEntry;
```

Comment:
I'm _very_ surprised CI hasn't gotten angry here: no use of raw new/delete. Use shared/unique_ptr if you need heap allocation

Reply:
Looks like it checks for `delete` but not `new`.

Reply:
TIL.  I'll work that in!

Reply:
Ah, I remember now, we still use a lot of raw `new` in the packet system and I haven't had time to fix it yet. The point is that if you `new` up memory, you're going to leak it if you don't eventually and consistently `delete` it later on. Smart pointers handle this for you when they fall out of scope (or their ref count reaches zero). 

It's important here because it's in a packet handler, so a sufficiently motivated bad actor could send in many bad packets, make the server allocate a bunch of memory that'll never be free'd, and then starve the process/machine of resources

Reply:
and as a general rule about C++, initialisation is scary and you should be careful! I'll try and explain as best I can:

```cpp
// default init
// default constructor will be called, but member vars won't be default initialised (despite the name), and so will be left uninitialised - typically you'd never use this. Uninitialised memory is garbage, remember.
// You've manually assigned all the members of FishingContestEntry a default value, so those will be used, but this isn't a typical use-case.
auto a = new Object;

// value init
// default constructor will be called, and member vars will be value initialised depending on their types - which is much safer.
auto a = new Object();
```

Reply:
Behold, nightmares: https://en.cppreference.com/w/cpp/language/initialization